### PR TITLE
Refactoring on python deps generation

### PR DESCRIPTION
**Pull request by ocristian**

### CODE REVIEW

File: plugins/dependency-checker/src/main/java/com/freenow/sauron/plugins/generator/python/PythonDependencyGenerator.java [hunk 1]
```diff
@@ -49,36 +44,26 @@ public Path generateCycloneDxBom(Path repositoryPath)
...
         try
         {
-            createPythonVirtualEnv(repositoryPath);
             generateRequirementsFreeze(repositoryPath);
-
-            return repositoryPath.resolve("bom.xml");
+            Path resolved = repositoryPath.resolve("bom.xml");
+            if (resolved.toFile().exists() && Files.size(resolved) > 0)
+            {
+                log.info("BOM file created.");
```
Comment (sergioasantiago) [inline plugins/dependency-checker/src/main/java/com/freenow/sauron/plugins/generator/python/PythonDependencyGenerator.java:51]: can you log the file path?

File: plugins/dependency-checker/src/main/java/com/freenow/sauron/plugins/generator/python/PythonDependencyGenerator.java [hunk 2]
```diff
@@ -49,36 +44,26 @@ public Path generateCycloneDxBom(Path repositoryPath)
...
+            }
+            else
+            {
+                log.info("BOM file is either empty or does not exist.");
```
Comment (sergioasantiago) [inline plugins/dependency-checker/src/main/java/com/freenow/sauron/plugins/generator/python/PythonDependencyGenerator.java:55]: Can you log the result of the resolved.toFile().exists() and Files.size(resolved)

File: sauron-service/Dockerfile [hunk 3]
```diff
@@ -53,14 +53,27 @@ RUN curl -s https://pyenv.run | bash && eval "$(/root/.pyenv/bin/pyenv init -)"
...
 RUN python -m pip install --upgrade pip
 
-# Enabling Poetry to be used without need of full path 
-RUN ln -s ~/.pyenv/versions/"$PYTHON_VERSION"/bin/poetry /usr/local/bin/poetry
+# Installing pipx - https://pipx.pypa.io/stable/
+RUN python -m pip install --user pipx
+RUN python -m pipx ensurepath
+
+# Installing Poetry
+RUN python -m pipx install poetry
```
Comment (sergioasantiago) [inline sauron-service/Dockerfile:64]: I would pin the version here, to avoid newer version to break the build

File: sauron-service/Dockerfile [hunk 4]
```diff
@@ -53,14 +53,27 @@ RUN curl -s https://pyenv.run | bash && eval "$(/root/.pyenv/bin/pyenv init -)"
...
+
+# Installing Poetry Export Plugin
+RUN python -m pipx inject poetry poetry-plugin-export
```
Comment (sergioasantiago) [inline sauron-service/Dockerfile:67]: I would pin the version here, to avoid newer version to break the build

File: sauron-service/Dockerfile [hunk 5]
```diff
@@ -53,14 +53,27 @@ RUN curl -s https://pyenv.run | bash && eval "$(/root/.pyenv/bin/pyenv init -)"
...
+
+# Installing CycloneDX Python SBOM Generation Tool
+RUN python -m pipx install cyclonedx-bom
```
Comment (sergioasantiago) [inline sauron-service/Dockerfile:70]: I would pin the version here, to avoid newer version to break the build

**Review by sergioasantiago — CHANGES_REQUESTED** [review]

File: plugins/dependency-checker/src/main/java/com/freenow/sauron/plugins/generator/python/PythonDependencyGenerator.java (hunk 2 above)
Comment (ocristian) [reply plugins/dependency-checker/src/main/java/com/freenow/sauron/plugins/generator/python/PythonDependencyGenerator.java:55]: we keep as is as at this phase the file does not exists or is empty.

**Pull request merged**